### DISCUSSION TITLE
[RFC] Bulk operations on library

### DISCUSSION THREAD

**Discussion by rajatmodi62** [stashapp/stash#3983] [Q&A] [answered]
Hi all, 
assuming that i have a big library, and all the perpetual hashes have been generated, 
[1] is there a way to bulk match them to the scenes, 
[2] all the tags of the scenes, should get imported in the library 
[3] same for all the performers. 
basically, i dont want to waste time manually selecting each scene, tag, or performer photo since it becomes cumbersome in a large library, 
is there any particular way to do it, 
thanks

**Comment by Flashy78** [comment]
https://docs.stashapp.cc/beginner-guides/guide-to-scraping/

**Comment by rajatmodi62** [comment]
this does not talk about inmporting all performers and tags automatically, 
i still have to go and add each tag

**Comment by stg-annon** [comment, marked as answer]
@rajatmodi62  there are various ways to do what you want the tool built into Stash that you're looking for is called the [Identify task](https://docs.stashapp.cc/in-app-manual/tasks/identify/),

⚠️ Warning⚠️ 
- it is not perfect and this may lead to you having a bunch of improperly matched scenes
- Be sure to make a backup of your db first since theres no undo button in Stash

the task will auto create performers and tags from the stash-box/fragment scrapers that it matches against based on how you configure it for each one

**Comment by Charlotte-br560** [comment]
You can automate bulk operations using Python libraries like Pandas or PySpark for matching hashes to scenes. Consider using [Crawlbase](https://crawlbase.com/) for extracting scene tags and performer info.
